File: csharp/MaxPainUI/ClientApp/src/app/models/chart-info.ts
export class ChartInfo {
    chartType: String;
    dataType: String;
    title: String;
    hAxisTitle: String;
    hAxisFormat: String;
    vAxisTitle: String;
    vAxisFormat: String;
    width: Number;
    height: Number;
    interval: Number;
    enable3D: boolean;
    isDarkMode: boolean;
    isTransparent: boolean;
    viewMin: 1;
    viewMax: 100;

    series: ChartSeries [] = [];

    dataArray : Array<{strike: Number, call: Number, put: Number}>;
}

export class ChartSeries {
    title: String;
    color: String;

    points: ChartPoint [] = [];
}

export class ChartPoint {
    x: Number;
    y: Number;
}

File: csharp/MaxPainChart/MaxPainChart/Models/ChartInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MaxPainChart.Models
{
    public class ChartInfo
    {
        public string ChartType { get; set; }
        public string Title { get; set; }
        public string HAxisTitle { get; set; }
        public string HAxisFormat { get; set; }
        public string VAxisTitle { get; set; }
        public string VAxisFormat { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public int Interval { get; set; }
        public bool Enable3D { get; set; }
        public bool IsDarkMode { get; set; }
        public bool IsTransparent { get; set; }

        public List<ChartSeries> Series { get; set; }

        public List<object> DataArray { get; set; }

        public ChartInfo()
        {
            Series = new List<ChartSeries>();
            Width = 1200;
            Height = 600;
            Interval = 5;
            Enable3D = false;
        }
    }

    public class ChartSeries
    {
        public string Title { get; set; }
        public string Color { get; set; }
        public List<ChartPoint> Points { get; set; }

        public ChartSeries()
        {
            Points = new List<ChartPoint>();
        }
    }

    public class ChartPoint
    {
        public double X { get; set; }
        public double Y { get; set; }

        public ChartPoint()
        {
        }
        public ChartPoint(double x, double y)
        {
            X = x;
            Y = y;
        }
    }
}